app/Http/Controllers/PostController.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use \App\Models\Post;
+use Image;
+
+class PostController extends Controller
+{
+    public function createPost() {
+        return view('home');
+    }
+    public function posts() {
+        $posts = Post::all();
+        return view('posts', ['posts' => $posts]);
+    }
+
+    public function create(Request $request, $user_id) {
+
+        $request->validate(
+            [
+                'name' => 'required|max:20',
+                'description' => 'required|max:255',
+                'image' => 'required|mimes:jpg,png,jpeg|max:5048'
+            ],
+            [
+                'name.max:20' => 'Reduce size of title',
+                'password.max:255' => 'Reduce size of body'
+            ]
+        );
+
```

Comment:
Given that user_id is user input, this must also be validated.

In the case of a user ID, if the user is already authenticated you would access user_id from `$request->user()->id` or `auth()->id`. 

I would be very wary about trusting a user ID that is provided via user input in most circumstances.